src/rtpi.h

```diff
@@ -12,9 +12,19 @@
 
 #include "rtpi_internal.h"
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 typedef union pi_mutex pi_mutex_t;
 typedef union pi_cond pi_cond_t;
 
+#ifdef __cplusplus
```

Comment:
I'm still working my way through this PR but I've noticed this. I think we can actually get rid of the 'restrict' qualifier because there's no danger of pointer aliasing here. I have a commit in my 'latest' branch I should have upstreamed but never got around to it: https://github.com/gratian/librtpi/commit/bbd6bd8907c5f37224843306218548013dcef1ab

I can post a PR for it if we think that's a better option.

Reply:
Yeah, if we can get rid of `restrict` entirely that feels like a cleaner solution. Should I wait for you to PR those changes, or would you prefer I just grab that commit to include in the next round of this PR?

Reply:
I've posted a PR for removing the `restrict` qualifier: https://github.com/dvhart/librtpi/pull/26